packages/frontend/lib/index.ts

```diff
@@ -377,6 +379,14 @@ export default class Nango {
             return { params: tableauCredentials } as unknown as ConnectionConfig;
         }
 
+        if ('api_key' in credentials) {
+            const perimeterCredentials: PerimeterCredentials = {
+                api_key: credentials.api_key
+            };
+
+            return { params: perimeterCredentials } as unknown as ConnectionConfig;
+        }
```

Comment:
this works only because no other credentials has a attribute called api_key in snake case. It is very fragile and might be a source of bugs if more credentials with an api_key attribute are added. This PR is probably not the place to refactor but a good refactoring of this function would be welcome. For instance credentials could validated via a zod schema